src/Acl.php

```diff
@@ -58,6 +55,14 @@ class Acl implements AclInterface
      */
     protected $resources = [];
 
+    /**
+     * Resources by resourceId plus a null element
+     * Used to speed up setRule()
+     *
+     * @var (ResourceInterface|null)[]
+     */
+    protected $resourcesById = [null];
```

Comment:
Since this is a new property, let's make it `private` to prevent abuse

Reply:
Also, we do this for speed, at the cost of memory usage: best to keep this concept hidden/internal 👍